src/routes/Apps/App/TrainDialogs.tsx

```diff
@@ -102,51 +102,34 @@ function getColumns(intl: InjectedIntl): IRenderableColumn[] {
     let equalizeColumnWidth = window.innerWidth / 6
     return [
         {
-            key: 'firstInput',
-            name: Util.formatMessageId(intl, FM.TRAINDIALOGS_FIRSTINPUT),
-            fieldName: 'firstInput',
+            key: `tags`,
+            name: `Tags`,
+            fieldName: `Tags`,
             minWidth: 100,
             maxWidth: equalizeColumnWidth,
             isResizable: true,
             isSortedDescending: true,
-            render: trainDialog => {
-                let firstInput = getFirstInput(trainDialog);
-                if (firstInput) {
-                    return (<span className={textClassName(trainDialog)} data-testid="train-dialogs-first-input">
-                        {trainDialog.validity && trainDialog.validity !== CLM.Validity.VALID &&
-                            <OF.Icon
-                                className={`cl-icon ${ValidityUtils.validityColorClassName(trainDialog.validity)}`}
-                                iconName="IncidentTriangle"
-                            />
-                        }
-                        {firstInput}
-                    </span>)
-                }
-                return <OF.Icon iconName="Remove" className="notFoundIcon" />
-            },
-            getSortValue: trainDialog => {
-                let firstInput = getFirstInput(trainDialog)
-                return firstInput ? firstInput.toLowerCase() : ''
-            }
+            render: trainDialog => <TagsReadOnly data-testid="train-dialogs-tags" className={textClassName(trainDialog)} tags={trainDialog.tags} />,
+            getSortValue: trainDialog => trainDialog.tags.join(' ')
         },
         {
-            key: 'lastInput',
-            name: Util.formatMessageId(intl, FM.TRAINDIALOGS_LASTINPUT),
-            fieldName: 'lastInput',
+            key: `description`,
+            name: `Description`,
+            fieldName: 'description',
             minWidth: 100,
             maxWidth: equalizeColumnWidth,
             isResizable: true,
             render: trainDialog => {
-                let lastInput = getLastInput(trainDialog)
-                if (lastInput) {
-                    return <span className={textClassName(trainDialog)} data-testid="train-dialogs-last-input">{lastInput}</span>
-                }
-                return <OF.Icon iconName="Remove" className="notFoundIcon" />
+                const firstInput = getFirstInput(trainDialog)
+                const lastInput = getLastInput(trainDialog)
+                return <>
+                    <span data-testid="train-dialogs-description" className={textClassName(trainDialog)}>{trainDialog.description || `Open dialog to add description`}</span>
+                    {/* Keep firstInput and lastInput here until tests are upgraded */}
+                    <span style={{ visibility: "hidden", position: "absolute" }} data-testid="train-dialogs-first-input">{firstInput ? firstInput : ''}</span>
```

Comment:
@SDESkowronski 

My first comment somehow go attached to review and is now stale, but here is what I was suggesting.  Preserves the content and `data-testid` attributes in hidden elements.

Reply:
You did good on this change! I got the branch and tried a couple of my tests that rely on this and they work...but then you probably already knew that.

I'm not sure that we will want to rely on this long term, but it is a good way to bridge the gap.